packages/go_router/lib/src/matching.dart

```diff
@@ -71,14 +71,20 @@ class RouteMatchList {
   /// Removes the last match.
   void pop() {
     _matches.removeLast();
-    assert(
-        _matches.isNotEmpty,
-        'You have popped the last page off of the stack,'
-        ' there are no pages left to show');
+
+    _assertNotEmpty();
```

Comment:
will probably be good to wrap this in an assert too

Reply:
and the name should be _debugAssertNotEmpty

Reply:
I renamed it in c70f33cb85522bb8a503f9c5a8ee92a1006bc522, but I'm not sure why we should wrap the invocation in an assert?